Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/EditorModeOutline.azsl

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <EditorModeCommon.azsli>
+#include <EditorModeDepthTransition.azsli>
+
+partial ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
+{    
+    float m_lineThickness;
+    float4 m_lineColor;
+}
+
+//! Applies an outline effect to the entity mask of the specified line thickness.
+//! @note There are known issues with pimples due to z-fighting of coincident surfaces in the entity mask that will be addressed it LYN-10304.
+float selectedEntityOutliner(const float2 uv, const float lineThickness)
+{
+    const float2 maskSize = PassSrg::GetImageSize(PassSrg::m_entityMask);
+    const float2 unormalizedUv = float2(uv.x * maskSize.x, uv.y * maskSize.y);
+    const float2 maskSizeReciprocal = float2(1., 1.0) / maskSize;
+    float r = 0.;
+
+    // Selected entity mask value
+    float S = PassSrg::m_entityMask.Sample(PassSrg::PointSampler, unormalizedUv * maskSizeReciprocal).a;
```

Comment:
Can you rename r and S to something more descriptive? Like outlineAmount and maskValue?

Reply:
Fixed 👍